utils/system_locale/src/linux.rs

```diff
@@ -0,0 +1,79 @@
+#[cfg(target_os = "linux")]
+mod linux_locale {
+    use locale::{
```

Comment:
I don't think we want to depend on the `locale` crate, we're kind of the rewrite it wants to do: https://github.com/rust-locale/rust-locale/issues/7 (this crate is the `langinfo_sys` part). @zbraniecki do we want to revive the discussions in `rust-locale` at this time?

`set_locale` is available in [`libc`](https://docs.rs/libc/0.2.155/libc/fn.setlocale.html), and the constants you should define yourself.